crates/driver/src/infra/observe/mod.rs

```diff
@@ -18,21 +18,24 @@ use {
             quote::{self, Quote},
             Liquidity,
         },
+        infra::{mempool, solver},
         util::http,
     },
     url::Url,
 };
 
 mod metrics;
 
-// TODO The idea is that some of the functions below will also update the
-// metrics. This should be fairly easy to do, just go down the list and think
-// about if the thing is worth measuring or not. This will be done in a
-// follow-up PR.
+/// Setup the observability. The log argument configures the tokio tracing
+/// framework.
+pub fn init(log: &str) {
+    boundary::initialize_tracing(log);
+    metrics::init();
+}
 
 /// Observe a received auction.
-pub fn auction(auction: &Auction) {
-    tracing::info!(?auction, "received auction");
+pub fn auction(solver: &solver::Name, auction: &Auction) {
+    tracing::info!(%solver, ?auction, "received auction");
```

Comment:
I think we should try it like this for a bit. Not using spans has the advantage of being more functional/explicit. There is no hidden global state with the span. For example, this won't break if we move the future to a tokio::task, but a span would break. Either approach has up and down sides.

Reply:
It's literally global state vs. local parameters.

Reply:
LOL, that seems like a bad faith summary of the argument.
I would have expected you to be the first person to complain when types have to capture data they don't actually  care about just because we want to have nice logs.

Reply:
Yeah you're probably right

Reply:
Let me give that another try

Reply:
Either way, I don't think this bikeshed is worth any more discussion

Reply:
I will keep it the way it is and suffer the consequences if this approach doesn't work, or alternatively reap the benefits if it turns out to be the best solution in the long term

Reply:
The only way to tell is to try it and give it some time and see. At least I'm doing something to improve the situation with logging, maybe it's not worth it, I don't know